frontend/src/main/java/com/google/tripmeout/frontend/serialization/GsonPlaceVisitTypeAdapter.java

```diff
@@ -0,0 +1,90 @@
+package com.google.tripmeout.frontend.serialization;
+
+import com.google.appengine.repackaged.com.google.api.client.json.Json;
+import com.google.common.base.Strings;
+import com.google.gson.JsonParseException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.PlaceVisitModel.UserMark;
+import java.io.IOException;
+
+public class GsonPlaceVisitTypeAdapter extends TypeAdapter<PlaceVisitModel> {
+  private static final String PLACE_VISIT_MODEL_PLACE_ID_JSON_FIELD_NAME = "placeId";
+  private static final String PLACE_VISIT_MODEL_TRIP_ID_JSON_FIELD_NAME = "tripId";
+  private static final String PLACE_VISIT_MODEL_NAME_JSON_FIELD_NAME = "name";
+  private static final String PLACE_VISIT_MODEL_USER_MARK_JSON_FIELD_NAME = "userMark";
+  private static final String PLACE_VISIT_MODEL_LATITUDE_JSON_FIELD_NAME = "latitude";
+  private static final String PLACE_VISIT_MODEL_LONGITUDE_JSON_FIELD_NAME = "longitude";
+
+  @Override
+  public PlaceVisitModel read(JsonReader reader) throws IOException {
+    if (reader.peek() == JsonToken.NULL) {
+      reader.nextNull();
+      return null;
+    }
+    PlaceVisitModel.Builder placeBuilder = PlaceVisitModel.builder();
+    reader.beginObject();
+    while (reader.hasNext()) {
+      JsonToken token = reader.peek();
+      if (token.equals(JsonToken.NAME)) {
```

Comment:
I may be missing something, but what happens here if the reader _has_ a token within this object, but it's _not_ a name? Does this just loop indefinitely? (I would expect `reader.hasNext()` and `reader.peek()` to not update the reader itself.)

I would recommend just calling `reader.nextName()` here and letting an `IOException` escape if that's not what you have.